flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -256,117 +256,106 @@ public WebRuntimeMonitor(
-			.GET("/jobs/:jobid/yarn-stop", handler(new JobStoppingHandler()))
-
-			// DELETE is the preferred way of stopping a job (Rest-conform)
-			.DELETE("/jobs/:jobid/stop", handler(new JobStoppingHandler()));
+		// stop a job via GET (for proper integration with YARN this has to be performed via GET)
+		GET(router, new JobStoppingHandler());
+		// DELETE is the preferred way of stopping a job (Rest-conform)
+		DELETE(router, new JobStoppingHandler());
 
 		int maxCachedEntries = config.getInteger(
 				ConfigConstants.JOB_MANAGER_WEB_CHECKPOINTS_HISTORY_SIZE,
 			ConfigConstants.DEFAULT_JOB_MANAGER_WEB_CHECKPOINTS_HISTORY_SIZE);
 		CheckpointStatsCache cache = new CheckpointStatsCache(maxCachedEntries);
 
 		// Register the checkpoint stats handlers
-		router
-			.GET("/jobs/:jobid/checkpoints", handler(new CheckpointStatsHandler(currentGraphs)))
-			.GET("/jobs/:jobid/checkpoints/config", handler(new CheckpointConfigHandler(currentGraphs)))
-			.GET("/jobs/:jobid/checkpoints/details/:checkpointid", handler(new CheckpointStatsDetailsHandler(currentGraphs, cache)))
-			.GET("/jobs/:jobid/checkpoints/details/:checkpointid/subtasks/:vertexid", handler(new CheckpointStatsDetailsSubtasksHandler(currentGraphs, cache)));
+		GET(router, new CheckpointStatsHandler(currentGraphs));
+		GET(router, new CheckpointConfigHandler(currentGraphs));
+		GET(router, new CheckpointStatsDetailsHandler(currentGraphs, cache));
+		GET(router, new CheckpointStatsDetailsSubtasksHandler(currentGraphs, cache));
 
 		if (webSubmitAllow) {
-			router
-				// fetch the list of uploaded jars.
-				.GET("/jars", handler(new JarListHandler(uploadDir)))
+			// fetch the list of uploaded jars.
+			GET(router, new JarListHandler(uploadDir));
 
-				// get plan for an uploaded jar
-				.GET("/jars/:jarid/plan", handler(new JarPlanHandler(uploadDir)))
+			// get plan for an uploaded jar
+			GET(router, new JarPlanHandler(uploadDir));
 
-				// run a jar
-				.POST("/jars/:jarid/run", handler(new JarRunHandler(uploadDir, timeout, config)))
+			// run a jar
+			POST(router, new JarRunHandler(uploadDir, timeout, config));
 
-				// upload a jar
-				.POST("/jars/upload", handler(new JarUploadHandler(uploadDir)))
+			// upload a jar
+			POST(router, new JarUploadHandler(uploadDir));
 
-				// delete an uploaded jar from submission interface
-				.DELETE("/jars/:jarid", handler(new JarDeleteHandler(uploadDir)));
+			// delete an uploaded jar from submission interface
+			DELETE(router, new JarDeleteHandler(uploadDir));
 		} else {
-			router
-				// send an Access Denied message (sort of)
-				// Every other GET request will go to the File Server, which will not provide
-				// access to the jar directory anyway, because it doesn't exist in webRootDir.
-				.GET("/jars", handler(new JarAccessDeniedHandler()));
+			// send an Access Denied message
+			JarAccessDeniedHandler jad = new JarAccessDeniedHandler();
+			GET(router, jad);
+			POST(router, jad);
```

Comment:
Because there are Jar-related handlers that accept POST and DELETE requests. Having the JarAccessDeniedHandler reject these as well seemed more consistent.